core/src/main/java/tech/pegasys/web3signer/core/FilecoinRunner.java

```diff
@@ -65,11 +63,8 @@ protected String getOpenApiSpecResource() {
   }
 
   @Override
-  protected Router populateRouter(final Context context) {
-    final ArtifactSignerProvider signerProvider =
-        loadSigners(config, context.getVertx(), context.getMetricsSystem());
-    incSignerLoadCount(context.getMetricsSystem(), signerProvider.availableIdentifiers().size());
-
+  protected Router populateRouter(
+      final Context context, final ArtifactSignerProvider signerProvider) {
```

Comment:
Can getArtifactSignerProvider method be used instead of this additional argument?

Reply:
unfortunately not, because we need reference of signerProvider outside populateRouter (for closing the resources).